### DISCUSSION TITLE
What exactly is in the release tarball

### DISCUSSION THREAD

**Discussion by IndrekHaav** [lldap/lldap#348] [Q&A] [answered]
I wanted to test this without Docker, so I downloaded the tarball from the releases page, expecting it to contain a compiled, ready-to-run version of the app. And while there's indeed an `lldap` executable in there and `./lldap run` seems to bring up a web server on port 17170, there's a bunch of errors in the console about `./app/pkg` and `./app/static` missing (which I understand are front-end assets). Opening `localhost:17170` also gives an error "No such file or directory (os error 2)".

At the same time, the contents of the tarball don't seem to be enough to compile the project, either - there's no cargo.toml nor an `app` directory. So how exactly is one supposed to use the release tarballs?

**Comment by nitnelave** [comment, marked as answer]
Hmm, the tarballs may not be the most... tested part of the release :D 
It seems that the frontend files are in the wrong folder (index.html, main.js and pkg should be in a subfolder called `app`), and it's missing the `static` folder (which you can just get from git). Not an ideal experience, I recognize.

Thankfully, the issue has already been fixed: https://github.com/nitnelave/lldap/pull/325

In the next release, you'll have a direct link to the lldap binary compiled for most popular (linux) architectures, and a link to the zipped web resources that you can just unzip next to it.

**Comment by IndrekHaav** [comment]
Thanks for the quick response! I got it to work with the following steps:

```shell
$ mkdir lldap
$ tar xzvf lldap-x86_64-v0.4.1.tar.gz --strip-components=3 -C lldap/
$ cd lldap
$ mkdir app data
$ mv -t app/ index.html main.js pkg
$ cp -R ../lldap_src/app/static/ app/
```

I also renamed `lldap_config.docker_template.toml` to `lldap_config.toml`, edited the username/password and removed the leading slash from the database and server key filenames. After that, `./lldap run` brought up the web UI.
